out_writeapi.go

```diff
@@ -284,16 +284,22 @@ func sendRequestExactlyOnce(ctx context.Context, data [][]byte, config **outputC
 func sendRequestRetries(ctx context.Context, data [][]byte, config **outputConfig, streamIndex int) error {
 	retryer := newStatelessRetryer((*config).numRetries)
 	attempt := 0
-	currStream := (*(*config).managedStreamSlice)[streamIndex]
 	for {
 		err := sendRequestExactlyOnce(ctx, data, config, streamIndex)
 		if err == nil {
 			break
 		}
 		//unsuccesful data append
 		if rebuildPredicate(err) {
-			currStream.managedstream.Finalize(ctx)
-			currStream.managedstream.Close()
+			streamSlice := (*config).managedStreamSlice
+			// finalizes all streams
+			for i := 0; i < len(*streamSlice); i++ {
+				currStream := (*streamSlice)[i]
+				currStream.managedstream.Finalize(ctx)
```

Comment:
How are we making sure that the currStream doesn't have any active requests on it before we finalize and close it?

Reply:
I'm not really sure. This would be more of a question for Raahi rather than me.

Reply:
If this PR is creating a regression for the exactly-once scenario, then it will need to be addressed here too. First we need to understand how the failed stream is behaving and whether it's doing the right thing or not, then we need to figure out the correct behavior for the other streams and make sure we implement that too if needed. Please consult with Raahi and let me know.

Reply:
Remove for loop, change buildStream parameters